### PR TITLE
feat: support organizations

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- support organizations
- add react sample code for organizations
- refactor `useLogto()` hook
- update types and constants

## Testing

- added a ton of unit tests
- tested react sample, successfully get organization token

image

## Checklist

- [x] `.changeset`
- [x] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/client/src/index.ts [hunk 1]
```diff
@@ -151,7 +181,8 @@ export default class LogtoClient {
    */
   async fetchUserInfo(): Promise<UserInfoResponse> {
     const { userinfoEndpoint } = await this.getOidcConfig();
-    const accessToken = await this.getAccessToken();
+    // eslint-disable-next-line unicorn/no-useless-undefined
+    const accessToken = await this.getAccessToken(undefined, undefined);
```
Comment (wangsijie) [inline packages/client/src/index.ts:185]: What is the purpose of these two undefined?

**Review by wangsijie — APPROVED** [review]

File: packages/client/src/index.ts [hunk 2]
```diff
@@ -134,7 +152,19 @@ export default class LogtoClient {
    * resource, as specified in the Logto Console.
    */
   async getAccessTokenClaims(resource?: string): Promise<AccessTokenClaims> {
-    const accessToken = await this.getAccessToken(resource);
+    // eslint-disable-next-line unicorn/no-useless-undefined
+    const accessToken = await this.getAccessToken(resource, undefined);
```
Comment (charIeszhao) [inline packages/client/src/index.ts:156]: Why the explicit `undefined`?

**Review by charIeszhao — APPROVED** [review]
Looks good to me

File: packages/react-sample/src/consts/index.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export const baseUrl = window.location.origin;
 export const redirectUrl = `${baseUrl}/callback`;
 
-export const appId = 'sampleId'; // Register the sample app in Logto dashboard
-export const endpoint = 'http://localhost:3001'; // Replace with your own Logto endpoint
+export const appId = 'gos7awdecqru059iptvc5'; // Register the sample app in Logto dashboard
+export const endpoint = 'https://k95znw.app.logto.dev/'; // Replace with your own Logto endpoint
```
Comment (charIeszhao) [inline packages/react-sample/src/consts/index.ts:5]: nit: Remember to remove these changes

File: packages/client/src/index.ts (hunk 2 above)
Comment (gao-sun) [reply packages/client/src/index.ts:156]: was trying to use `undefined` for more clear function calling, but removed now.

File: packages/client/src/index.ts (hunk 1 above)
Comment (gao-sun) [reply packages/client/src/index.ts:185]: was trying to use `undefined` for more clear function calling, but removed now.

**Pull request merged**